src/libopensc/card-openpgp.c

```diff
@@ -317,6 +318,19 @@ struct pgp_priv_data {
 	sc_security_env_t	sec_env;
 };
 
+static int
+get_full_pgp_aid(sc_card_t *card, sc_file_t *file)
+{
+	int r = 0;
+	/* explicitly get the full aid */
+	r = sc_get_data(card, 0x004F, file->name, sizeof file->name);
+	if (r < 0)
+		file->namelen = 0;
+	else
+		file->namelen = r;
```

Comment:
This can be simplified as `file->namelen = MAX(r, 0)`, which might be even more readable.